apiserver/facades/agent/uniter/uniter.go

```diff
@@ -134,8 +136,31 @@ func (u *UniterAPI) EnsureDead(ctx context.Context, args params.Entities) (param
 			result.Results[i].Error = apiservererrors.ServerError(err)
 			continue
 		}
-		if err := u.applicationService.EnsureUnitDead(ctx, unitName, u.leadershipRevoker); err != nil {
+		unitUUID, err := u.applicationService.GetUnitUUID(ctx, unitName)
+		if errors.Is(err, applicationerrors.UnitNotFound) {
+			result.Results[i].Error = apiservererrors.ParamsErrorf(params.CodeNotFound, "unit %q not found", unitName)
```

Comment:
Why not just use `apiservererrors.ServerError(jujuerrors.NotFoundf("unit %q", unitName))`?

Same in the other places?

Reply:
This pattern was recommended to me by @tlm. I think because it avoids the import of `jujuerrors`

I have no preference personally